### ISSUE TITLE
Version 2023.5.31 - Basic external control via messaging: First test results

### ISSUE THREAD

**Issue by LilywhiteLilith** [snowie2000/mactype#940]
@snowie2000 

(Sorry for opening a new issue, it really should have gone in a different place: #934)

Thank you very much for the quick implementation!

Here are my first test results.

Tested under the following OS:

- Windows 7 Pro
- Windows 8.1 Pro
- Windows 10 Pro
- Windows 11 Pro

(Main user with admin rights)

MacType not installed via setup, just unzipped ZIP archive into a folder.
MacType runs in "manual" mode (!)

MacType is started by my program via shell command.
In the MacType.ini the program writes one of 3 profiles via API command (WritePrivateProfileString) as "AlternativeFile".
After that a restart of MacTray is triggered via shell command ("/s restart").
The profile changes are now visible in the program!

Enable, disable, stop, restart: everything works fine!

:-)

Custom profile icons:
Unfortunately the MacTray icon is not updated, it does not show the custom icon of the activated profile.
When I open the popup menu of MacTray with the mouse, the correct profile is always selected. But the default icon of MacTtray is always shown.

A change of the icon occurs only...

...when i open the popup menu, then write another profile into the MacType.ini and then start MacTray with /s restart. Now the MacTray icon changes! BUT: it now shows the icon of the PREVIOUS profile, i.e. the profile BEFORE the profile change.

OK, it is a small, optical "blemish", with which I can live. But if it should be possible, that also the MacTry icon changes to the custom profile icon of the activated profile after a restart via shell command...
...that would be fantastic of course!!!

Thanks a lot for the great work! MacType is a real jewel!!!

Lilywhite Lilith

**Comment by snowie2000** [comment]
What a keen eye!

I checked the code and you're absolutely right! Actually, the profile list and icons are only updated when you popup the context menu.

Check this out: [MacTray-icon-fix.zip](https://github.com/snowie2000/mactype/files/11620060/MacTray-icon-fix.zip)

**Comment by LilywhiteLilith** [comment]
Yes, now the display of the profile icons after a profile change is correct. Everything works fine now!
Excellentissimo!
:-)

**Comment by LilywhiteLilith** [comment]
One more little thing caught my keen eye.
Immediately after starting MacTray.exe the icon of the default profile is always displayed (or the MacTray.exe icon). It would be nice, if the icon of the profile, which is found as "AlternativeFile", would be displayed right after the start?
I hope I'm not getting on your nerves with my "keen eye"!
;-)

**Comment by snowie2000** [comment]
Mactype tray mode always starts disabled, then it scans profiles and auto enables itself if it was set to.

When it was designed, I didn't find a way to gray out icons, so the only icon I can display on startup is the built-in disable icon.

**Comment by LilywhiteLilith** [comment]
This is something I can live with very well, no real problem. Thanks for your explanation!
